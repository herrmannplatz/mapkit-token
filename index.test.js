/* eslint-env jest */
const fs = require('fs')
const jwt = require('jsonwebtoken')
const generate = require('./')

// mock date
Date.now = jest.fn(() => 0)

describe('generate', () => {
  it('should generate a token', () => {
    const authKey = fs.readFileSync('./key.p8')
    const teamId = 'A1A1A1A1A1'
    const keyId = 'B1B1B1B1B1'
    const ttl = 60 * 60
    const origin = 'org.team.id'

    const token = generate(authKey, keyId, teamId, ttl, origin)
    expect(jwt.decode(token)).toMatchSnapshot()
  })

  it('should generate a token without ttl and origin', () => {
    const authKey = fs.readFileSync('./key.p8')
    const teamId = 'A1A1A1A1A1'
    const keyId = 'B1B1B1B1B1'

    const token = generate(authKey, keyId, teamId)
    expect(jwt.decode(token)).toMatchSnapshot()
  })

  it('should throw in case of missing parameters', () => {
    const authKey = fs.readFileSync('./key.p8')
    const teamId = 'A1A1A1A1A1'
    const keyId = 'B1B1B1B1B1'

    expect(() => generate()).toThrowError()
    expect(() => generate(authKey)).toThrowError()
    expect(() => generate(authKey, keyId)).toThrowError()
    expect(() => generate(authKey, keyId, teamId)).not.toThrowError()
  })
})
