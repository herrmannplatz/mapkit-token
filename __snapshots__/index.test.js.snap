// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate should generate a token 1`] = `
Object {
  "exp": 3600,
  "iat": 0,
  "iss": "A1A1A1A1A1",
  "origin": "org.team.id",
}
`;

exports[`generate should generate a token without ttl and origin 1`] = `
Object {
  "exp": 1800,
  "iat": 0,
  "iss": "A1A1A1A1A1",
}
`;
